File: plugin/src/main/java/co/uk/magmo/puretickets/tasks/ReminderTask.java
package co.uk.magmo.puretickets.tasks;

import co.uk.magmo.puretickets.interactions.NotificationManager;
import co.uk.magmo.puretickets.locale.Messages;
import co.uk.magmo.puretickets.ticket.TicketManager;
import co.uk.magmo.puretickets.ticket.TicketStatus;
import co.uk.magmo.puretickets.utilities.Constants;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;
import org.bukkit.scheduler.BukkitRunnable;

public class ReminderTask extends BukkitRunnable {
    private final TicketManager ticketManager;
    private final NotificationManager notificationManager;

    public ReminderTask(TicketManager ticketManager, NotificationManager notificationManager) {
        this.ticketManager = ticketManager;
        this.notificationManager = notificationManager;
    }

    @Override
    public void run() {
        Integer amount = ticketManager.count(TicketStatus.OPEN);

        if (amount == 0) return;

        for (Player player : Bukkit.getOnlinePlayers()) {
            if (player.hasPermission(Constants.STAFF_PERMISSION + ".remind")) {
                notificationManager.basic(player, Messages.OTHER__REMINDER, "%amount%", amount.toString());
            }
        }
    }
}

File: src/main/kotlin/co/uk/magmo/puretickets/tasks/ReminderTask.kt
package co.uk.magmo.puretickets.tasks

import co.uk.magmo.puretickets.interactions.NotificationManager
import co.uk.magmo.puretickets.locale.Messages
import co.uk.magmo.puretickets.ticket.TicketManager
import co.uk.magmo.puretickets.ticket.TicketStatus
import co.uk.magmo.puretickets.utils.Constants
import org.bukkit.Bukkit
import org.bukkit.scheduler.BukkitRunnable

class ReminderTask(private val ticketManager: TicketManager, private val notificationManager: NotificationManager) : BukkitRunnable() {
    override fun run() {
        val amount = ticketManager.asMap().values.flatten()
                .count { it.status == TicketStatus.OPEN }

        if (amount == 0) return

        Bukkit.getOnlinePlayers()
                .filter { it.hasPermission(Constants.STAFF_PERMISSION + ".remind") }
                .forEach { notificationManager.reply(it, Messages.TASKS__REMINDER, "%amount%", amount.toString()) }
    }
}